src/foam/core/reflow/PropertyExprView.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.core.reflow',
+  name: 'PropertyExprView',
+  extends: 'foam.u2.View',
+
+  requires: [ 'foam.core.reflow.PropertyChoiceView' ],
+
+  properties: [
+    {
+      name: 'data',
+      documentation: 'The data to be transformed',
+      preSet: function(oldValue, newValue) {
+        if ( ! newValue ) {
+          // Clear selections when data is null/undefined
+          this.selectedProperty = null;
+          this.selectedTransformation = null;
+          return newValue;
+        }
+        
+        // Check if this is a transformation expression (has delegate)
+        if ( newValue.delegate ) {
+          // This is a transformation expression
+          this.selectedTransformation = newValue.cls_;
+          this.selectedProperty = newValue.delegate;
+        } else {
+          // This is just a plain property
+          this.selectedProperty = newValue;
+          this.selectedTransformation = null;
+        }
+        
+        return newValue;
+      }
+    },
+    {
+      name: 'forCls',
+      documentation: 'The class to get properties from'
+    },
+    {
+      name: 'selectedProperty',
+      documentation: 'The currently selected property'
+    },
```

Comment:
Don't you need a postSet on this property which updates data? If the new Property is a different class then you should reset the selectedTransformation also.